zcash_client_backend/src/data_api.rs

```diff
@@ -1391,12 +1391,13 @@ pub trait InputSource {
     /// specified shielded protocol.
     ///
     /// Returns `Ok(None)` if the note is not known to belong to the wallet or if the note
-    /// is not spendable.
+    /// is not spendable as of the given height.
     fn get_spendable_note(
         &self,
         txid: &TxId,
         protocol: ShieldedProtocol,
         index: u32,
+        target_height: TargetHeight,
```

Comment:
If we're going to make this change with the given rationale (that an `InputSource` can't know what the "chain tip" is: https://github.com/zcash/librustzcash/pull/1987#discussion_r2422223178), we should be consistent.
- Update the `InputSource` trait documentation to make clear that implementations of it must not assume a particular chain tip height.
- Either add a `target_height` field to `InputSource::get_account_metadata` (which must be used to decide whether notes are spent and thus excluded), or documentation how it should handle the above requirement that it not assume a particular chain height.
- Either add a `target_height` field to `InputSource::get_unspent_transparent_output`, or (given that it is almost a pure coin getter) move it to `WalletRead`.